src/litdata/streaming/downloader.py

```diff
@@ -203,6 +209,56 @@ def download_bytes(self, remote_filepath: str, offset: int, length: int, local_c
 
         return response["Body"].read()
 
+    def download_fileobj(self, remote_filepath: str, fileobj: Any) -> None:
+        """Download a file from S3 directly to a file-like object."""
+        obj = parse.urlparse(remote_filepath)
+
+        if obj.scheme != "s3":
+            raise ValueError(f"Expected obj.scheme to be `s3`, instead, got {obj.scheme} for remote={remote_filepath}")
+
+        if not hasattr(self, "_client"):
+            self._client = S3Client(storage_options=self._storage_options, session_options=self.session_options)
+
+        bucket = obj.netloc
+        key = obj.path.lstrip("/")
+
+        self._client.client.download_fileobj(
+            bucket,
+            key,
+            fileobj,
+        )
+
+    def _get_store(self, bucket: str):
+        """Return an obstore S3Store instance for the given bucket, initializing if needed."""
+        if not hasattr(self, "_store"):
+            if not _OBSTORE_AVAILABLE:
+                raise ModuleNotFoundError(str(_OBSTORE_AVAILABLE))
+            from obstore.auth.boto3 import Boto3CredentialProvider
+            from obstore.store import S3Store
+
+            session = boto3.Session(**self._storage_options, **self.session_options)
+            credential_provider = Boto3CredentialProvider(session)
+            self._store = S3Store(bucket, credential_provider=credential_provider)
+        return self._store
+
+    async def adownload_fileobj(self, remote_filepath: str, fileobj: Any) -> None:
```

Comment:
I think it is better to change the API here to take the remote_filepath and return the bytes directly.